packages/a/assimp/xmake.lua

```diff
@@ -0,0 +1,98 @@
+    add_configs("ignore_git_hash",       {description = "Don't call git to get the hash", default = false, type = "boolean"})
+    add_configs("install_pdb",           {description = "Install MSVC debug files", default = false, type = "boolean"})
+
+    add_deps("cmake")
+
+    on_load(function (package)
+        if is_plat("windows", "linux", "macosx") then
+            if package:config("build_samples") then
+                package:add("deps", "freeglut")
+            end
+        end
+        if is_plat("linux") or package:config("build_zlib") then
+            package:add("deps", "zlib")
+        end
+    end)
+
+    on_install("windows", "linux", "macosx", function (package)
+        local config = {}
+
+        local function add_config_arg(config_name, cmake_name)
+            table.insert(config, "-D" .. cmake_name .. "=" .. (package:config(config_name) and "ON" or "OFF"))
+        end
+
+        add_config_arg("shared",               "BUILD_SHARED_LIBS")
+        add_config_arg("double_precision",     "ASSIMP_DOUBLE_PRECISION")
+        add_config_arg("opt_build_packages",   "ASSIMP_OPT_BUILD_PACKAGES")
+        add_config_arg("no_export",            "ASSIMP_NO_EXPORT")
+        add_config_arg("build_zlib",           "ASSIMP_BUILD_ZLIB")
+        add_config_arg("build_samples",        "ASSIMP_BUILD_SAMPLES")
+        add_config_arg("build_tests",          "ASSIMP_BUILD_TESTS")
+        add_config_arg("coveralls",            "ASSIMP_COVERALLS")
+        add_config_arg("werror",               "ASSIMP_WERROR")
+        add_config_arg("asan",                 "ASSIMP_ASAN")
+        add_config_arg("ubsan",                "ASSIMP_UBSAN")
+        add_config_arg("system_irrxml",        "SYSTEM_IRRXML")
+        add_config_arg("build_docs",           "BUILD_DOCS")
+        add_config_arg("ignore_git_hash",      "IGNORE_GIT_HASH")
+
+        if is_plat("macosx") then
+            add_config_arg("build_framework", "BUILD_FRAMEWORK")
+        end
+
+        if is_plat("windows") then
+            table.insert(config, "-DASSIMP_INSTALL_PDB=" .. ((package:debug() or package:config("install_pdb")) and "ON" or "OFF"))
+        end
+
+        if is_plat("linux", "macosx") then
+            table.insert(config, "-DINJECT_DEBUG_POSTFIX=" .. ((package:debug() or package:config("inject_debug_postfix")) and "ON" or "OFF"))
+        end
+
+        if is_plat("android") then
+            add_config_arg("android_jniiosysystem", "ASSIMP_ANDROID_JNIIOSYSTEM")
+        end
+
+        if is_plat("windows", "linux", "macosx") then
+            add_config_arg("build_assimp_tools", "ASSIMP_BUILD_ASSIMP_TOOLS")
+        end
+
+        if is_plat("android", "iphoneos") then
+            table.insert(config, "-DASSIMP_BUILD_ASSIMP_TOOLS=OFF")
+        end
+
+        import("package.tools.cmake").install(package, config)
+        if is_plat("linux") then
+            package:add("links", "assimp", "IrrXML")
```

Comment:
you can uses add_syslinks("irrXML") to ensure link order.

Reply:
The irrXML library is built into assimp and gets compiled with it when ```system_irrxml``` is false. Which is the default behaviour.